pkg/detectors/make/api_token/api_token.go

```diff
@@ -0,0 +1,140 @@
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+	detectors.DefaultMultiPartCredentialProvider
+	detectors.EndpointSetter
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+var _ detectors.EndpointCustomizer = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"make"}) + `\b([a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12})\b`)
+	// Pattern to match Make.com URLs in the data
+	urlPat = regexp.MustCompile(`\b(eu|us)[12]\.make\.(com|celonis\.com)`)
+)
+
+func (Scanner) CloudEndpoint() string { return "" }
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"make.com", "make.celonis.com"}
+}
+
+// FromData will find and optionally verify Make secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	uniqueMatches := make(map[string]struct{})
+	for _, match := range keyPat.FindAllStringSubmatch(dataStr, -1) {
+		uniqueMatches[match[1]] = struct{}{}
+	}
+
+	// Extract Make URLs from the data
+	foundURLs := urlPat.FindAllString(dataStr, -1)
+
+	// Get endpoints using EndpointCustomizer
+	endpoints := s.Endpoints(foundURLs...)
+
+	// Skip creating results if no endpoints are available
+	if len(endpoints) == 0 {
+		return
+	}
+
+	for match := range uniqueMatches {
+		// Create results for each endpoint
+		for _, endpoint := range endpoints {
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_MakeApiToken,
+				Raw:          []byte(match),
+				RawV2:        []byte(match + ":" + endpoint),
+			}
+
+			if verify {
+				client := s.client
+				if client == nil {
+					client = defaultClient
+				}
+
+				isVerified, extraData, verificationErr := verifyMatch(ctx, client, match, endpoint)
+				s1.Verified = isVerified
+				s1.ExtraData = extraData
+				s1.SetVerificationError(verificationErr, match)
+			}
+
+			results = append(results, s1)
+		}
+	}
+
+	return
+}
+
+func verifyMatch(ctx context.Context, client *http.Client, token string, endpoint string) (bool, map[string]string, error) {
+	verified, err := tryURL(ctx, client, fmt.Sprintf("https://%s/api/v2/", endpoint), token)
+	if verified {
+		return true, nil, nil
+	}
+	if err != nil {
+		// Indeterminate failure (network error, etc.)
+		return false, nil, err
+	}
+	// Determinate failure (401)
+	return false, nil, nil
+}
+
+func tryURL(ctx context.Context, client *http.Client, baseURL, token string) (bool, error) {
+	// This endpoint returns 200 OK if the token is valid and the correct FQDN for the API is used.
```

Comment:
A couple of suggestions:

1. We can simplify the code by keeping just a single `VerifyMatch` function. It can handle making the request, sending it, and processing the response just like we usually do.
2. The second return value is always `nil`, so we can remove it entirely.